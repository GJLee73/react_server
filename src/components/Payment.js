/**
 * 결제페이지. 아임포트를 사용. TODO: 카카오페이 같은 결제 모듈도 사용할 수 있게.
 * @author G1
 * @logs // 18.2.25
 */

import React from 'react';

class Payment extends React.Component {

	componentWillMount(){
		window.IMP.init('imp60269147');
	}

	render(){
		return(
			<div>
				<form>
					<input type='BUTTON' defaultValue='pay' onClick={this.requestPay}/>
				</form>
			</div>
		)
	}

	requestPay(){
		window.IMP.request_pay({
			pg : 'nice', // version 1.1.0부터 지원.
			pay_method : 'card',
			merchant_uid : 'merchant_' + new Date().getTime(),
			name : 'test_name',
			amount : 1000,
			buyer_email : 'iamport@siot.do',
			buyer_name : 'dfiodiowe',
			buyer_tel : '010-1234-5678',
			buyer_addr : '서울특별시 강남구 삼성동',
			buyer_postcode : '123-456',
			m_redirect_url : 'https://www.naver.com'
			}, function(rsp) {
			if ( rsp.success ) {
			    var msg = '결제가 완료되었습니다.';
			    msg += '고유ID : ' + rsp.imp_uid;
			    msg += '상점 거래ID : ' + rsp.merchant_uid;
			    msg += '결제 금액 : ' + rsp.paid_amount;
			    msg += '카드 승인번호 : ' + rsp.apply_num;
			} else {
				msg = '결제에 실패하였습니다.';
			    msg += '에러내용 : ' + rsp.error_msg;
			}
			alert(msg);
		});
	}
}

export default Payment;