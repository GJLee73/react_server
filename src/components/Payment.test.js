import React from 'react';
import {shallow} from 'enzyme'
import Payment from './Payment'

describe('Payment test',()=>{
    var requestPay=jest.spyOn(Payment.prototype,'requestPay')
    
    global.alert=jest.fn()
    global.IMP={init:jest.fn(),request_pay:jest.fn((a,callback)=>{callback('')})}
    var wrapper = shallow(<Payment/>)
    describe('requestPay test',()=>{
       
        it('success',()=>{   
            var rsp = {success:true, imp_uid:'1', merchant_uid:'2',paid_amount:'3',apply_num:'4'}
            global.IMP={init:jest.fn(),request_pay:jest.fn((a,callback)=>{callback(rsp)})}
        var wrapper = shallow(<Payment/>)
            wrapper.find('input').simulate('click')
            expect(requestPay.mock.calls.length).toBe(1)
            expect(global.alert.mock.calls[0][0]).toBe("결제가 완료되었습니다.고유ID : 1상점 거래ID : 2결제 금액 : 3카드 승인번호 : 4")
        })
        it('failure',()=>{   
            var rsp = {success:false, error_msg:'실패'}
            global.IMP={init:jest.fn(),request_pay:jest.fn((a,callback)=>{callback(rsp)})}
        var wrapper = shallow(<Payment/>)
            wrapper.find('input').simulate('click')
            expect(requestPay.mock.calls.length).toBe(2)
            expect(global.alert.mock.calls[1][0]).toBe( "결제에 실패하였습니다.에러내용 : 실패")
        })
    })

    it('snapshot test',()=>{
        expect(wrapper).toMatchSnapshot()
    })
})