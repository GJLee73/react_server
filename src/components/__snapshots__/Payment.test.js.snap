// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Payment test snapshot test 1`] = `
<div>
  <form>
    <input
      defaultValue="pay"
      onClick={[Function]}
      type="BUTTON"
    />
  </form>
</div>
`;
